### PR TITLE
[i18n-ar] Translated file : `docs/source/ar/philosophy.md` into Arabic

### PR DESCRIPTION
**Pull request by AhmedAlmaghz**
## What does this PR do?
Translated the `docs/source/ar/philosophy.md` file of the documentation to `Arabic العربية`. I use transformers for my side-project and wanted to contribute. As a beginner, this seemed appropriate.

### Some questions:

- I can't access the pr documentation endpoint for `Arabic العربية`. Is there a missing step?
- I left all the other files alone after copying the English docs. Is this acceptable for future updates or should I add contents to the `_toctree.yml` only when translation is complete? Should I delete all other files for a cleaner review?
- In `Arabic`, people use both `Transformer` and `المحولات` to describe the model. However, as this is more of a product of HuggingFace I opted not to translate it. May you please let me know which you would prefer?

Thank you in advance for your review.

Part of #32435

### Before submitting

- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests), Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/) ?Please add a link to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

 ### Who can review?
 @stevhliu , @MKhalusova , @abodacs, may you please review this PR?

### CODE REVIEW

**Comment by AhmedAlmaghz** [comment]
@abodacs

Next Review :
# CONCEPTUAL GUIDES

- [ ]  Philosophy #33064
- [x]  Glossary #33038
- [ ]  What 🤗 Transformers can do #33073
- [ ]  How 🤗 Transformers solve tasks #33074
- [ ]  The Transformer model family #33047
- [ ]  Summary of the tokenizers #33078
- [ ]  Attention #33021
- [ ]  Padding and truncation #33050
- [ ]  BERTology #33024
- [ ]  Perplexity of fixed-length models #33063
- [ ]  Pipelines for webserver inference #33066
- [ ]  Model training anatomy #33045
- [ ] Getting the most out of LLMs #33043

File: docs/source/ar/philosophy.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
+# الفلسفة
+
+🤗 Transformers هي مكتبة موجهة تم بناؤها من أجل:
```
Comment (abodacs) [inline docs/source/ar/philosophy.md:3]:
```suggestion
تُعد 🤗 Transformers مكتبة برمجية ذات رؤية واضحة صُممت من أجل:```

File: docs/source/ar/philosophy.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+- باحثي وأخصائيي التعليم في مجال التعلم الآلي الذين يسعون إلى استخدام نماذج المحولات واسعة النطاق أو دراستها أو توسيعها.
+- الممارسون العمليون الذين يرغبون في ضبط تلك النماذج أو تشغيلها في الإنتاج، أو كليهما.
+- المهندسون الذين يريدون فقط تنزيل نموذج مُدرب مسبقًا واستخدامه لحل مهمة تعلم آلي معينة.
```
Comment (abodacs) [inline docs/source/ar/philosophy.md:7]:
```suggestion
- باحثي والمُتعلّمون في مجال التعلم الآلي ممن يسعون لاستخدام أو دراسة أو تطوير نماذج Transformers واسعة النطاق.
- مُطبّقي تعلم الآلة الذين يرغبون في ضبط تلك النماذج أو تشغيلها في بيئة إنتاجية، أو كليهما.
- المهندسون الذين يريدون فقط تنزيل نموذج مُدرب مسبقًا واستخدامه لحل مهمة تعلم آلي معينة.
```

File: docs/source/ar/philosophy.md [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+تم تصميم المكتبة مع الأخذ في الاعتبار هدفين رئيسيين:
+
+1. أن تكون سهلة وسريعة الاستخدام قدر الإمكان:
```
Comment (abodacs) [inline docs/source/ar/philosophy.md:11]:
```suggestion
1.سهولة وسرعة الاستخدام:
```

File: docs/source/ar/philosophy.md [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+
+  - قمنا بالحد بشكل كبير من عدد التجريدات المواجهة للمستخدم والتي يجب تعلمها، وفي الواقع، لا توجد تجريدات تقريبًا،
+    فقط ثلاث فئات قياسية مطلوبة لاستخدام كل نموذج: [التكوين](main_classes/configuration)،
+    [نماذج](main_classes/model)، وفئة ما قبل المعالجة ([معالج](main_classes/tokenizer) لـ NLP، [معالج الصور](main_classes/image_processor) للرؤية، [مستخرج الميزات](main_classes/feature_extractor) للصوت، و [معالج](main_classes/processors) للإدخالات متعددة الوسائط).
```
Comment (abodacs) [inline docs/source/ar/philosophy.md:15]:
```suggestion
  - تمّ تقليل عدد المفاهيم المُجردة التي يتعامل معها المستخدم إلى أدنى حد والتي يجب تعلمها، وفي الواقع، لا توجد مفاهيم مُجردة تقريبًا، فقط ثلاث فئات أساسية مطلوبة لاستخدام كل نموذج: [الإعدادات](main_classes/configuration)،
    [نماذج](main_classes/model)، وفئة ما قبل المعالجة ([مُجزّئ لغوي](main_classes/tokenizer) لـ NLP، [معالج الصور](main_classes/image_processor) للرؤية، [مستخرج الميزات](main_classes/feature_extractor) للصوت، و [معالج](main_classes/processors) للمدخﻻت متعددة الوسائط).
```

File: docs/source/ar/philosophy.md [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+  - يمكن تهيئة جميع هذه الفئات بطريقة بسيطة وموحدة من مثيلات مُدربة مسبقًا باستخدام طريقة `from_pretrained()` الشائعة والتي تقوم بتنزيل (إذا لزم الأمر)، وتخزين
+    وتحميل مثيل الفئة المرتبطة وبياناتها (معلمات التهيئة، ومعجم المعالج،
```
Comment (abodacs) [inline docs/source/ar/philosophy.md:17]:
```suggestion
  - يمكن تهيئة جميع هذه الفئات بطريقة بسيطة وموحدة من خلال نماذج مُدربة مسبقًا باستخدام الدالة الموحدة  `from_pretrained()`  والتي تقوم بتنزيل (إذا لزم الأمر)، وتخزين وتحميل  كل من: فئة النموذج المُراد استخدامه والبيانات المرتبطة ( مُعاملات الإعدادات، ومعجم للمُجزّئ اللغوي،
```

File: docs/source/ar/philosophy.md [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+    وأوزان النماذج) من نقطة تفتيش مُدربة مسبقًا مقدمة على [Hugging Face Hub](https://huggingface.co/models) أو نقطة التفتيش المحفوظة الخاصة بك.
```
Comment (abodacs) [inline docs/source/ar/philosophy.md:18]:
```suggestion
    وأوزان النماذج) من نقطة  تدقيق مُحددة مُخزّنة على [Hugging Face Hub](https://huggingface.co/models) أو ن من نقطة تخزين خاصة بالمستخدم.
```

File: docs/source/ar/philosophy.md [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+  - بالإضافة إلى هذه الفئات الأساسية الثلاث، توفر المكتبة واجهتي برمجة تطبيقات: [`pipeline`] للاستدلال
+    بسرعة باستخدام نموذج لمهمة معينة و [`Trainer`] لتدريب أو ضبط نموذج PyTorch بسرعة (جميع نماذج TensorFlow متوافقة مع `Keras.fit`).
```
Comment (abodacs) [inline docs/source/ar/philosophy.md:20]:
```suggestion
  - بالإضافة إلى هذه الفئات الأساسية الثلاث، توفر المكتبة واجهتي برمجة تطبيقات: [`pipeline`] للاستخدام السريع لأحد النماذج لأداء استنتاجات على مهمة مُحددة، و [`Trainer`] للتدريب السريع أو الضبط الدقيق لنماذج PyTorch  (جميع نماذج TensorFlow متوافقة مع `Keras.fit`).
```

File: docs/source/ar/philosophy.md (hunk 1 above)
Comment (abodacs) [reply docs/source/ar/philosophy.md]: أسف أصفت بالخطأ  ثﻻث غﻻمات ```
@AhmedAlmaghz

**Review by stevhliu — COMMENTED** [review]
LGTM, just remember to add it to the `toctree` :)

File: docs/source/ar/philosophy.md [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+- **فئات ما قبل المعالجة** تحويل البيانات الخام إلى تنسيق مقبول من قبل النموذج. يقوم [المعالج](main_classes/tokenizer) بتخزين المعجم لكل نموذج ويقدم طرقًا لتشفير وفك تشفير السلاسل في قائمة من مؤشرات تضمين الرموز ليتم إطعامها للنموذج. تقوم [معالجات الصور](main_classes/image_processor) بمعالجة إدخالات الرؤية، وتقوم [مستخلصات الميزات](main_classes/feature_extractor) بمعالجة إدخالات الصوت، ويقوم [المعالج](main_classes/processors) بمعالجة الإدخالات متعددة الوسائط.
+
+يمكن تهيئة جميع هذه الفئات من مثيلات مُدربة مسبقًا، وحفظها محليًا، ومشاركتها على Hub باستخدام ثلاث طرق:
+
+- تسمح لك `from_pretrained()` بتهيئة نموذج وفئة تكوين وفئة ما قبل المعالجة من إصدار مُدرب مسبقًا إما يتم توفيره بواسطة المكتبة نفسها (يمكن العثور على النماذج المدعومة على [Model Hub](https://huggingface.co/models)) أو مخزن محليًا (أو على خادم) بواسطة المستخدم.
+- تسمح لك `save_pretrained()` بحفظ نموذج وفئة تكوين وفئة ما قبل المعالجة محليًا بحيث يمكن إعادة تحميله باستخدام `from_pretrained()`.
+- تسمح لك `push_to_hub()` بمشاركة نموذج وفئة تكوين وفئة ما قبل المعالجة على Hub، بحيث يمكن الوصول إليها بسهولة من قبل الجميع.
```
Comment (abodacs) [inline docs/source/ar/philosophy.md:49]:
```suggestion
- تسمح لك الدالة  `from_pretrained()` بتهيئة النموذج وتكويناته وفئة المعالجة المسبقة من إصدار مُدرب مسبقًا إما يتم توفيره بواسطة المكتبة نفسها (يمكن العثور على النماذج المدعومة على [Model Hub](https://huggingface.co/models)) أو مخزنة محليًا (أو على خادم) بواسطة المستخدم.
- تسمح لك الدالة  `save_pretrained()` بحفظ النموذج، وتكويناته وفئة المعالجة المسبقة محليًا، بحيث يمكن إعادة تحميله باستخدام الدالة `from_pretrained()`.
- تسمح لك `push_to_hub()` بمشاركة نموذج وتكويناتهوفئة المعالجة المسبقة على Hub، بحيث يمكن الوصول إليها بسهولة من قبل الجميع.
```

File: docs/source/ar/philosophy.md [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+  - طرق بسيطة لتنميق رؤوس المحول وإزالتها.
+
+- التبديل بسهولة بين PyTorch و TensorFlow 2.0 و Flax، مما يسمح بالتدريب باستخدام إطار واحد والاستدلال باستخدام إطار آخر.
+
+## المفاهيم الرئيسية
+
+تم بناء المكتبة حول ثلاثة أنواع من الفئات لكل نموذج:
+
+- **فئات النماذج** يمكن أن تكون نماذج PyTorch ([torch.nn.Module](https://pytorch.org/docs/stable/nn.html#torch.nn.Module))، أو نماذج Keras ([tf.keras.Model](https://www.tensorflow.org/api_docs/python/tf/keras/Model))، أو نماذج JAX/Flax ([flax.linen.Module](https://flax.readthedocs.io/en/latest/api_reference/flax.linen/module.html)) التي تعمل مع الأوزان المُدربة مسبقًا المقدمة في المكتبة.
+- **فئات التكوين** تخزن معلمات التهيئة المطلوبة لبناء نموذج (مثل عدد الطبقات والحجم المخفي). أنت لست مضطرًا دائمًا إلى إنشاء مثيل لهذه الفئات بنفسك. على وجه الخصوص، إذا كنت تستخدم نموذجًا مُدربًا مسبقًا دون أي تعديل، فإن إنشاء النموذج سيهتم تلقائيًا بتهيئة التكوين (والذي يعد جزءًا من النموذج).
```
Comment (abodacs) [inline docs/source/ar/philosophy.md:42]:
```suggestion
- **فئات الإعداد** تخزن معلمات التهيئة المطلوبة لبناء نموذج (مثل عدد الطبقات وحجم  الطبقة المخفية). أنت لست مضطرًا دائمًا إلى إنشاء مثيل لهذه الفئات بنفسك. على وجه الخصوص، إذا كنت تستخدم نموذجًا مُدربًا مسبقًا دون أي تعديل، فإن إنشاء النموذج سيهتم تلقائيًا تهيئة الإعدادات (والذي يعد جزءًا من النموذج).
```

File: docs/source/ar/philosophy.md [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+  - نتيجة لذلك، هذه المكتبة ليست صندوق أدوات متعدد الاستخدامات من الكتل الإنشائية للشبكات العصبية. إذا كنت تريد توسيع أو البناء على المكتبة، فما عليك سوى استخدام Python و PyTorch و TensorFlow و Keras العادية والوراثة من الفئات الأساسية للمكتبة لإعادة استخدام الوظائف مثل تحميل النموذج وحفظه. إذا كنت ترغب في معرفة المزيد عن فلسفة الترميز لدينا للنماذج، فراجع منشور المدونة الخاص بنا [Repeat Yourself](https://huggingface.co/blog/transformers-design-philosophy).
+
+2. تقديم نماذج رائدة في مجالها مع أداء قريب قدر الإمكان من النماذج الأصلية:
+
+  - نقدم مثالًا واحدًا على الأقل لكل بنية تقوم بإعادة إنتاج نتيجة مقدمة من المؤلفين الرسميين لتلك البنية.
+  - عادةً ما تكون الشفرة قريبة قدر الإمكان من قاعدة الشفرة الأصلية، مما يعني أن بعض شفرة PyTorch قد لا تكون "بأسلوب PyTorch" كما يمكن أن تكون نتيجة لكونها شفرة TensorFlow محولة والعكس صحيح.
+
+بعض الأهداف الأخرى:
+
+- عرض داخليات النماذج بشكل متسق قدر الإمكان:
```
Comment (abodacs) [inline docs/source/ar/philosophy.md:25]:
```suggestion
- كشف تفاصيل النماذج الداخلية بشكل متسق قدر الإمكان:
```

File: docs/source/ar/philosophy.md [hunk 11]
```diff
@@ -0,0 +1,49 @@
...
+
+  - نمنح حق الوصول، باستخدام واجهة برمجة تطبيقات واحدة، إلى جميع الحالات المخفية وأوزان الاهتمام.
+  - تم توحيد فئات ما قبل المعالجة وواجهات برمجة التطبيقات الأساسية للنموذج لتسهيل التبديل بين النماذج.
```
Comment (abodacs) [inline docs/source/ar/philosophy.md:28]:
```suggestion
  -نتيح الوصول، باستخدام واجهة برمجة واحدة، إلى جميع الحالات المخفية (Hidden-States) وأوزان الانتباه (Attention Weights).
  - تم توحيد واجهات برمجة التطبيقات الخاصة بفئات المعالجة المسبقة والنماذج الأساسية لتسهيل التبديل بين النماذج.

File: docs/source/ar/philosophy.md [hunk 12]
```diff
@@ -0,0 +1,49 @@
...
+
+- دمج مجموعة مختارة ذاتيًا من الأدوات الواعدة لضبط هذه النماذج ودراستها:
```
Comment (abodacs) [inline docs/source/ar/philosophy.md:30]:
```suggestion
- دمج مجموعة مختارة من الأدوات الواعدة لضبط النماذج بدقة (Fine-tuning) ودراستها:
```

File: docs/source/ar/philosophy.md [hunk 13]
```diff
@@ -0,0 +1,49 @@
...
+
+  - طريقة بسيطة ومتسقة لإضافة رموز جديدة إلى المعجم والمغلفات لضبط النماذج الدقيق.
+  - طرق بسيطة لتنميق رؤوس المحول وإزالتها.
```
Comment (abodacs) [inline docs/source/ar/philosophy.md:33]:
```suggestion
  - طريقة بسيطة ومتسقة لإضافة رموز جديدة إلى مفردات التضمينات (Embeddings) لضبط النماذج بدقة.
  - طرق سهلة لإخفاء (Masking) وتقليم (Pruning) رؤوس المحولات (Transformer Heads).
```

**Review by abodacs — CHANGES_REQUESTED** [review]
السﻻم عليكم

تمت المراجعة

@AhmedAlmaghz 

شكرا جزيﻻ

ملحوظة فى بعض الاحيان اراجع على فترات فاضع رسالة مثل هذه الرسالة

**Review by abodacs — APPROVED** [review]
@AhmedAlmaghz 
برجاء معالجة التعارض
